packages/git/src/browser/history/git-history-widget.ts

```diff
@@ -56,6 +58,30 @@ export class GitHistoryWidget extends GitBaseWidget<GitHistoryListNode> implemen
         this.addClass('theia-git');
     }
 
+    protected onAfterAttach(msg: Message) {
+        super.onAfterAttach(msg);
+        (async () => {
+            const sc = await this.getScrollContainer();
+            const listener = (e: UIEvent) => {
+                const el = (e.srcElement as HTMLElement);
+                if (el.scrollTop + el.clientHeight > el.scrollHeight - 83) {
```

Comment:
why `-83`?

Reply:
83 is the height of the header and the (mostly) invisible lazy load notification.
I know, magic numbers are not very good. I have to improve that.